pkg/distributor/distributor.go

```diff
@@ -660,6 +660,54 @@ func (d *Distributor) UserStats(ctx context.Context) (*UserStats, error) {
 	return totalStats, nil
 }
 
+// UserIDStats models ingestion statistics for one user, including the user ID
+type UserIDStats struct {
+	UserID string `json:"userID"`
+	UserStats
+}
+
+// AllUserStats returns statistics about all users.
+// Note it does not divide by the ReplicationFactor like UserStats()
+func (d *Distributor) AllUserStats(ctx context.Context) ([]UserIDStats, error) {
+	// Add up by user, across all responses from ingesters
+	perUserTotals := make(map[string]UserStats)
+
+	req := &client.UserStatsRequest{}
+	ctx = user.InjectOrgID(ctx, "1") // fake: ingester insists on having an org ID
+	// Not using d.forAllIngesters(), so we can fail after first error.
+	ingesters := d.ring.GetAll()
+	for _, ingester := range ingesters {
```

Comment:
Could use `forAllIngesters` here.

Reply:
look up 2 lines at the comment - if anything goes wrong I want to bail out not crash every ingester.